src/AccountManager.mo

```diff
@@ -331,54 +250,57 @@ module {
       return ?(inc, creditInc);
     };
 
-    // This function is async* but is guaranteed to never throw.
-    func processAllowance(p : Principal, account : ICRC1.Account, amount : Nat) : async* DrawResponse {
-      let res = await* Ledger.draw(p, account, amount);
-      if (R.isOk(res)) {
-        totalConsolidated_ += amount;
-        issue(p, amount);
+    // Proccesses allowance.
+    func processAllowance(p : Principal, account : ICRC1.Account, amount : Nat, expectedFee : ?Nat) : async* DepositFromAllowanceResponse {
+      switch (expectedFee) {
+        case null {};
+        case (?f) if (f != fee(#allowance)) return #err(#BadFee { expected_fee = fee(#allowance) });
```

Comment:
Isn't is it clearer to use `surcharge_` here in place of `fee(#allowance)` throughout the whole function? I know they are equal. But this code is the reason that they are equal. If they weren't equal then this code would be wrong.